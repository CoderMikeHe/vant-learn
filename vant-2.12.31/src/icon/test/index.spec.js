import Icon from '..';
import { mount } from '../../../test';

test('render icon with builtin icon name', () => {
  const wrapper = mount(Icon, {
    propsData: {
      name: 'success',
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('render icon with url name', () => {
  const wrapper = mount(Icon, {
    propsData: {
      name: 'https://img.yzcdn.com/icon.jpg',
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('render icon with local image', () => {
  const wrapper = mount(Icon, {
    propsData: {
      name: '/assets/icon.jpg',
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('render icon default slot', () => {
  const wrapper = mount({
    render(h) {
      return h(Icon, { props: { name: 'success' } }, ['Default slot']);
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('tag prop', () => {
  const wrapper = mount(Icon, {
    propsData: {
      tag: 'div',
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('dot prop', () => {
  const wrapper = mount(Icon, {
    propsData: {
      dot: true,
    },
  });
  expect(wrapper).toMatchSnapshot();
});

test('size without unit', () => {
  const wrapper = mount(Icon, {
    propsData: {
      size: 20,
    },
  });
  expect(wrapper).toMatchSnapshot();
});
