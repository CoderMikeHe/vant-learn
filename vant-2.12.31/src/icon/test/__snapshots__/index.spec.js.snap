// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dot prop 1`] = `
<i class="van-icon van-icon-undefined">
  <div class="van-info van-info--dot"></div>
</i>
`;

exports[`render icon default slot 1`] = `
<i class="van-icon van-icon-success">Default slot
  <!----></i>
`;

exports[`render icon with builtin icon name 1`] = `
<i class="van-icon van-icon-success">
  <!----></i>
`;

exports[`render icon with local image 1`] = `
<i class="van-icon"><img src="/assets/icon.jpg" class="van-icon__image">
  <!----></i>
`;

exports[`render icon with url name 1`] = `
<i class="van-icon"><img src="https://img.yzcdn.com/icon.jpg" class="van-icon__image">
  <!----></i>
`;

exports[`size without unit 1`] = `
<i class="van-icon van-icon-undefined" style="font-size: 20px;">
  <!----></i>
`;

exports[`tag prop 1`] = `
<div class="van-icon van-icon-undefined">
  <!---->
</div>
`;
